// files/jsheets/api/snippet_runtime_pb_service.js
// package: jsheets
// file: jsheets/api/snippet_runtime.proto

var jsheets_api_snippet_runtime_pb = require("../../jsheets/api/snippet_runtime_pb");
var grpc = require("@improbable-eng/grpc-web").grpc;

var SnippetRuntime = (function () {
  function SnippetRuntime() {}
  SnippetRuntime.serviceName = "jsheets.SnippetRuntime";
  return SnippetRuntime;
}());

SnippetRuntime.Evaluate = {
  methodName: "Evaluate",
  service: SnippetRuntime,
  requestStream: true,
  responseStream: true,
  requestType: jsheets_api_snippet_runtime_pb.EvaluateRequest,
  responseType: jsheets_api_snippet_runtime_pb.EvaluateResponse
};

exports.SnippetRuntime = SnippetRuntime;

function SnippetRuntimeClient(serviceHost, options) {
  this.serviceHost = serviceHost;
  this.options = options || {};
}

SnippetRuntimeClient.prototype.evaluate = function evaluate(metadata) {
  var listeners = {
    data: [],
    end: [],
    status: []
  };
  var client = grpc.client(SnippetRuntime.Evaluate, {
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport
  });
  client.onEnd(function (status, statusMessage, trailers) {
    listeners.status.forEach(function (handler) {
      handler({ code: status, details: statusMessage, metadata: trailers });
    });
    listeners.end.forEach(function (handler) {
      handler({ code: status, details: statusMessage, metadata: trailers });
    });
    listeners = null;
  });
  client.onMessage(function (message) {
    listeners.data.forEach(function (handler) {
      handler(message);
    })
  });
  client.start(metadata);
  return {
    on: function (type, handler) {
      listeners[type].push(handler);
      return this;
    },
    write: function (requestMessage) {
      client.send(requestMessage);
      return this;
    },
    end: function () {
      client.finishSend();
    },
    cancel: function () {
      listeners = null;
      client.close();
    }
  };
};

exports.SnippetRuntimeClient = SnippetRuntimeClient;
